lib/pacemaker/pcmk_graph_consumer.c

```diff
@@ -278,8 +278,8 @@ fire_synapse(crm_graph_t *graph, synapse_t *synapse)
             crm_err("Failed initiating <%s id=%d> in synapse %d",
                     crm_element_name(action->xml), action->id, synapse->id);
             pcmk__set_synapse_flags(synapse, pcmk__synapse_confirmed);
-            action->confirmed = TRUE;
-            action->failed = TRUE;
+            crm__set_graph_action_flags(action, crm__graph_action_confirmed);
+            crm__set_graph_action_flags(action, crm__graph_action_failed);
```

Comment:
You can combine these into one call (one of the benefits of the bitmask approach)